### PR TITLE
Check app update perms

### PR DESCRIPTION
**Pull request by rmweir**

### CODE REVIEW

File: pkg/api/customization/app/app.go [hunk 1]
```diff
@@ -94,14 +94,22 @@ func (w Wrapper) ActionHandler(actionName string, action *types.Action, apiConte
 		return err
 	}
 
+	var appMap map[string]interface{}
+	if err := access.ByID(apiContext, &projectschema.Version, projectv3.AppType, apiContext.ID, &appMap); err != nil {
+		return httperror.NewAPIError(httperror.NotFound, fmt.Sprintf("unable to access app by id: %v", err))
+	}
+
+	if err := apiContext.AccessControl.CanDo(pv3.AppGroupVersionKind.Group, pv3.AppResource.Name, "update", apiContext, appMap, apiContext.Schema); err != nil {
```
Comment (kinarashah) [inline pkg/api/customization/app/app.go]: q: why are we not using CanUpdate directly?

Comment (rmweir) [reply pkg/api/customization/app/app.go]: This is my understanding of the CanUpdate vs CanDo after speaking with Craig: CanUpdate returns nil if the schema is generally updatable, while CanDo considers rbac.

Comment (kinarashah) [reply pkg/api/customization/app/app.go]: oh, didn't know that, thank you!

File: pkg/api/customization/app/app.go [hunk 2]
```diff
@@ -94,14 +94,22 @@ func (w Wrapper) ActionHandler(actionName string, action *types.Action, apiConte
...
+		return httperror.NewAPIError(httperror.PermissionDenied, fmt.Sprintf("user does not have permission to update for action %s", actionName))
+	}
+
 	creatorNotFound := false
```
Comment (luthermonson) [inline pkg/api/customization/app/app.go]: probably dont need this any more. just move it in to one if kicking the error now that the compound conditional is removed.

Comment (rmweir) [reply pkg/api/customization/app/app.go]: reference in upgrade and rollback logic

**Comment by luthermonson** [comment]
lgtm

**Comment by cjellick** [comment]
lgtm

**Pull request merged**